transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,248 @@
+	pythonManage        = "manage"
+	pythonService       = "python"
+	djangoService       = "django"
+	requirementsTxtFile = "requirements.txt"
+	//DetectedPortsConfigType contains the detected ports
+	DetectedPortsConfigType transformertypes.ConfigType = "DetectedPortsConfigType"
+	//RequirementsTxtConfigType points to the requirements.txt file if it's present
+	RequirementsTxtConfigType transformertypes.ConfigType = "RequirementsTxtConfigType"
+	// DjangoProjectConfigType is set to true for Django projects
+	DjangoProjectConfigType transformertypes.ConfigType = "DjangoProjectConfigType"
+	// MainPythonFilesPathType points to the .py file path which contains main function
+	MainPythonFilesPathType transformertypes.PathType = "MainPythonFilesPathType"
+)
+
+var (
+	pythonMainRegex       = regexp.MustCompile(`__main__`)
+	djangoDependencyRegex = regexp.MustCompile(`django`)
+)
+
+// PythonDockerfileGenerator implements the Transformer interface
+type PythonDockerfileGenerator struct {
+	Config transformertypes.Transformer
+	Env    *environment.Environment
+}
+
+// PythonTemplateConfig implements python config interface
+type PythonTemplateConfig struct {
+	AppName           string
+	Port              int32
+	MainScriptRelPath string
+	RequirementsTxt   string
+	IsDjangoProject   bool
+}
+
+// Init Initializes the transformer
+func (t *PythonDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *PythonDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// findMainScripts returns the path of .py files having the main function
+func findMainScripts(pythonFilesPath []string) ([]string, error) {
+	if len(pythonFilesPath) == 0 {
+		return []string{}, nil
+	}
+	var pythonMainFiles []string
+	for _, pythonFilePath := range pythonFilesPath {
+		pythonFile, err := os.Open(pythonFilePath)
+		if err != nil {
+			logrus.Debugf("Failed to open the file %s", pythonFilePath)
+		}
+		scanner := bufio.NewScanner(pythonFile)
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			if pythonMainRegex.MatchString(scanner.Text()) {
+				pythonMainFiles = append(pythonMainFiles, pythonFilePath)
+			}
+		}
+	}
+	if len(pythonMainFiles) == 0 {
+		return []string{}, fmt.Errorf("could not find the main function in python files")
+	}
+	return pythonMainFiles, nil
+}
+
+// findDjangoDependency checks for django dependency in the requirements.txt file
+func findDjangoDependency(reqTxtFilePath string) bool {
+	reqTxtFile, err := os.Open(reqTxtFilePath)
+	if err != nil {
+		logrus.Debugf("Failed to open the file %s", reqTxtFilePath)
+	}
+	scanner := bufio.NewScanner(reqTxtFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		if djangoDependencyRegex.MatchString(scanner.Text()) {
+			return true
+		}
+	}
+	return false
```

Comment:
won't a ioutil.Readfile and strings.contains be sufficient?

Reply:
Done.